src/DenseACSets.jl

```diff
@@ -653,6 +734,68 @@ function replace_colons(acs::ACSet, parts::NamedTuple{types}) where {types}
   end)
 end
 
+# Garbage collection 
+####################
+
+"""
+Reindex the parts of the acset such that there are no gaps between the indices.
+Return a vector for each part mapping the new parts into the old parts. 
+"""
+function ACSetInterface.gc!(X::ACSet{<:MarkAsDeleted})
+  S = acset_schema(X)
+  μ = Dict(map(types(S)) do o 
+    p    = X.parts[o]
+    m    = collect(p.val)
+    m⁻¹  = Vector{Union{Int,Nothing}}(fill(nothing, p.next.x))
+    for (i, v) in enumerate(m)  
+      m⁻¹[v] = i 
+    end
+    return o => (m, m⁻¹)
+  end)
+  # Update homs and attrs
+  for (h, a, b) in homs(S)
+    μᵦ = μ[b][2]
+    X[h] = [μᵦ[X[a,h]] for a in μ[a][1]]
+    for i in (nparts(X,a)+1) : X.parts[a].next.x
+      clear_subpart!(X, i, h)
+    end
+  end
+  # TODO CLEAR HOM OF JUNK DATA
```

Comment:
Is this TODO up-to-date? If so and won't be resolved in this PR, it should get an issue. If it's no longer relevant, we should remove it.